server/src/main/java/org/opensearch/transport/InboundAggregator.java

```diff
@@ -128,7 +127,7 @@ public NativeInboundMessage finishAggregation() throws IOException {
         }
 
         final BreakerControl breakerControl = new BreakerControl(circuitBreaker);
-        final NativeInboundMessage aggregated = new NativeInboundMessage(currentHeader, releasableContent, breakerControl);
+        final ProtocolInboundMessage aggregated = new ProtocolInboundMessage(currentHeader, releasableContent, breakerControl);
```

Comment:
@reta The use of the NativeInboundMessage here really seemed wrong, because the aggregator doesn't care about the message payload at all, only the header. The issue is that a StreamInput is used to parse the header (in the `finishingParsingHeader()` method. My solution was to go head and expose a stream input in the protocol agnostic message type, which I think is fine. Similarly we'll need to expose the plain bytes to a future protobuf parser. The key is that knowledge of the structure of those bytes are only known to the appropriate protocol message handler in InboundHandler.

All of this is to say that I just discovered NativeInboundMessage only exists on main from PR #13126 since it was a breaking change. I think it might actually be simpler to revert that change (which is fair game since it was never backported).

Reply:
> @reta The use of the NativeInboundMessage here really seemed wrong, because the aggregator doesn't care about the message payload at all, only the header.

Yes, that make sense now I think. The aggregation is only based on size the payload (message), I think the previous thought process was to completely separate native and protobuf handling (assuming there is internal protocol knowledge in place so these classes could not be reused).  

> All of this is to say that I just discovered NativeInboundMessage only exists on main from PR https://github.com/opensearch-project/OpenSearch/pull/13126 since it was a breaking change. I think it might actually be simpler to revert that change (which is fair game since it was never backported).

Sure